src/Dao/CategoryDao.php

```diff
@@ -85,7 +94,16 @@ public function getCategoriesByParentId(string $parentid, int $languageId, int $
         return $categories;
     }
 
-    public function createCategory(Category $category, int $languageId, int $shopId): Category
+    public function saveCategory(Category $category, int $languageId, int $shopId): Category
+    {
+        if ($category->getId() === null) {
```

Comment:
I think we should come up with a better idea, because this leads to code duplication all the way down, as we have to methods `insertCategory` and `updateCategory` which are basically doing the same. Can we check if we could combine those two methods into one `saveCategoy` or `storeCategory` method? Or we could create two GraphQL mutations, one for creating and one for updating, which would make it more explicit two the client what is happening (as the client should know if he is creating or updating). What do you think?